src/wallet/spend.cpp

```diff
@@ -106,7 +106,13 @@ void CoinsResult::Erase(const std::unordered_set<COutPoint, SaltedOutpointHasher
 {
     for (auto& [type, vec] : coins) {
         auto remove_it = std::remove_if(vec.begin(), vec.end(), [&](const COutput& coin) {
-            return coins_to_remove.count(coin.outpoint) == 1;
+            // remove it if it's on the set
+            if (coins_to_remove.count(coin.outpoint) == 0) return false;
```

Comment:
in https://github.com/bitcoin/bitcoin/pull/26560/commits/1b14bd7ac77027351a1e5b3ef21fb2d1de659efd:

same comment from before, I'd suggest using `find` here.

Reply:
also, might be worth adding the `if () return false;` in the earlier commit so you don't end up touching this line twice?

Reply:
> in [1b14bd7](https://github.com/bitcoin/bitcoin/commit/1b14bd7ac77027351a1e5b3ef21fb2d1de659efd):
> 
> same comment from before, I'd suggest using `find` here.

answered in the first commit comment https://github.com/bitcoin/bitcoin/pull/26560#discussion_r1034897955


> also, might be worth adding the `if () return false;` in the earlier commit so you don't end up touching this line twice?

not sure what you meant, the first commit has a single line (the `count` function will either return 1 if the element is in the set or 0 if not).
```c++
return coins_to_remove.count(coin.outpoint) == 1;
```

Reply:
yeah, more just saying you could write the first commit like:

```
if (coins_to_remove.count(coin.outpoint) == 0) return false;
return true;
```

because you know that in a later commit you will add the `total_amount` and `total_effective` code, and this way you avoid re-writing the same line twice. it doesn't really matter in this case, just a suggestion. in general, it can be confusing as a reviewer when the same line of code is edited multiple times in different commits

Reply:
Ideally, commits should introduce atomic changes. Did the first one in this way so it can be backported without any dependency into a different branch (due the lack of https://github.com/bitcoin/bitcoin/pull/25685, v24 branch cannot cherry-pick https://github.com/bitcoin/bitcoin/commit/1b14bd7ac77027351a1e5b3ef21fb2d1de659efd).
So, we would only have that commit inside v24, which I think that is ugly.